students/BrandonHenson/Lesson9/Mailroom_5.py

```diff
@@ -0,0 +1,138 @@
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+    @property
+    def average(self):
+        return self.total/self.num_of_donations
+
+    @property
+    def num_of_donations(self):
+        return len(self.donations)
+
+
+class All(Donor):
+    def __init__(self):
+        pass
+
+    def all_total(self):
+        list_of_sums = []
+        count = 0
+        for k, v in Donor.donor_history.items():
+            for i in v:
+                count = count+1
+            total = sum(v)
+            list_of_sums.append(total)
+        gt = (sum(list_of_sums))
+        av = gt/count
+        gt = str(round(gt, 2))
+        av = str(round(av, 2))
+        print('The grand total of all donations is ${}.\
+ The average amount per donation is ${}.'.format(gt, av))
+
+
+def main_menu():
+    arg_dict = {
+        '1': menu_1,
+        '2': menu_2,
+        '3': email_all,
+        '4': exit
+    }
+
+    prompt = '\nSelect an option:\n' \
+             '[1] Send a Thank You\n' \
+             '[2] Create a Report\n' \
+             '[3] Send letters to everyone\n' \
+             '[4] Exit\n' \
+
+    menu_selection(prompt, arg_dict)
+
+
+def menu_selection(prompt, arg_dict):
+    try:
+        while True:
+            response = input(prompt)
+            arg_dict[response]()
+    except KeyError:
+        print("Enter 1,2,3, or 4")
+
+
+def menu_1():
+    try:
+        fullname = input("Enter a full name or 'list' to view all\n")
+        if fullname.lower() == 'list':
+            print(Donor.name_list)
+
+        elif str(fullname) in Donor.donor_history:
+            amount = int(input("Donation amount? \n"))
+            newvalue = Donor.donor_history.get(fullname) + [amount]
+            donor_history[fullname] = newvalue
+            print("Dear {}\nThank you for your generous donation of ${}\nIt will \
+     be put to good use.\n\n Sincerely,\n\
+            -The Team".format(fullname, amount))
+        elif str(fullname) not in Donor.donor_history:
+            amount = int(input("Donation amount? \n"))
+            Donor.donor_history[fullname] = amount
+            print("Dear {}\nThank you for your generous donation\
+     of ${}\nIt will be put to good use.\n\n Sincerely,\n\
+            -The Team".format(fullname, amount))
+    except ValueError:
+        print("Please enter a number for the amount.")
+
+
+def menu_2():
+    try:
+        print("\nDonor Name         |  Total Given | Num Gifts | Average Gift")
+        print("------------------------------------------------------------\n")
+        report = list()
+        for donor, total in Donor.donor_history.items():
+            report.append([donor, sum(total), len(total), sum(total)/len(total)])
+        sorted_report = sorted(report, key=lambda x: -x[1])
+
+        for i in sorted_report:
+            print("{:<20}   ${:>8} {:>8}        ${:<1}".format(
+                i[0], i[1], i[2], i[3]))
+        print(All.all_total(self=All))
```

Comment:
The `self` parameter is a special parameter provided by Python to give access to the instance of a class in a method definition. It will always be the first positional argument of an instance method, and you never want/need to explicitly set it. However, in this case, you're not instantiating the object or using the `self` parameter in the method definition, which is why this would be more appropriately defined as a static method